### PR TITLE
feat: changes needed to accommodate pets datasets.

### PR DESCRIPTION
**Pull request by sayakpaul**
The next PR will fix the Gradio template.

Related to #35.

### CODE REVIEW

**Comment by sayakpaul** [comment]
Why are we passing?

https://github.com/deep-diver/semantic-segmentation-ml-pipeline/blob/141c0947a2aff43a8f960d4119b0d9857ff26fec/training_pipeline/kubeflow_runner.py#L24-L25

Asking because we're already specifying the epochs.

**Comment by sayakpaul** [comment]
The pipeline is currently failing the training step and I couldn't know why from the logs on Vertex. 

A question:

* Why are using

https://github.com/deep-diver/semantic-segmentation-ml-pipeline/blob/141c0947a2aff43a8f960d4119b0d9857ff26fec/training_pipeline/models/signatures.py#L36

When the model's input name is actually different?

https://github.com/deep-diver/semantic-segmentation-ml-pipeline/blob/141c0947a2aff43a8f960d4119b0d9857ff26fec/training_pipeline/models/train.py#L96

**Comment by deep-diver** [comment]
> Why are we passing? Asking because we're already specifying the epochs.

yeah this parameters are not used currently.

**Comment by deep-diver** [comment]
> Also, help me understand why do we need this resizing? 

that part should be modified. we haven't modified it to suit to the different dataset(pets)

**Comment by deep-diver** [comment]
> Why do we need to transform the input feature names using transformed_name()?

this is not a requirement. I just did this to distinguish transformed and not-transformed features.

**Comment by deep-diver** [comment]
> return {CONCRETE_INPUT: decoded_images} 

oh, then this might be a mistake.

**Comment by sayakpaul** [comment]
@deep-diver for this PR for training, I would like to work on this branch. For the latter changes, I will work on a fork so that I can set up the secrets and run using my own GCP credits.

**Comment by sayakpaul** [comment]
Also, it seems like the Vertex AI training service is unable to find CUDA:

Screenshot 2022-10-30 at 10 43 36 AM

**Comment by sayakpaul** [comment]
Training seems to be working: https://console.cloud.google.com/vertex-ai/locations/us-central1/pipelines/runs/segmentation-training-pipeline-20221030053858.

**Comment by sayakpaul** [comment]
The Space and the HF model pusher seem to be working well. 

We need to take care of the rest of the concerns expressed in the PR. I will let you go through them once again and then we can discuss more. 

The main thing that needs to be resolved is the Gradio template.

File: training_pipeline/models/hyperparams.py [hunk 1]
```diff
@@ -1,10 +1,6 @@
 INPUT_IMG_SIZE = 128
 
-TRAIN_LENGTH = 800
-EVAL_LENGTH = 200
-
 TRAIN_BATCH_SIZE = 64
 EVAL_BATCH_SIZE = 64
 
-EPOCHS = 30
-LR = 0.001
```
Comment (deep-diver) [inline training_pipeline/models/hyperparams.py:10]: why omit LR ?

Comment (sayakpaul) [reply training_pipeline/models/hyperparams.py:10]: I am not trying to match the hyperparameters followed by the TensorFlow tutorial. They use the defaults.

**Review by deep-diver — APPROVED** [review]

**Comment by deep-diver** [comment]
> The main thing that needs to be resolved is the Gradio template.

need to modify Gradio template to work well on Pets dataset?

**Comment by sayakpaul** [comment]
> > The main thing that needs to be resolved is the Gradio template.
> 
> need to modify Gradio template to work well on Pets dataset?

Right! 

But there are other concerns that need to be addressed as well.

**Comment by deep-diver** [comment]
@sayakpaul 

let's list them up, and split the workloads!

**Comment by sayakpaul** [comment]
> @sayakpaul
> 
> let's list them up, and split the workloads!

They are enlisted as comments in this PR. Let me know if they are unclear.

**Comment by deep-diver** [comment]
- resizing
- transformed_name()
- {CONCRETE_INPUT: decoded_images}

right?

**Comment by sayakpaul** [comment]
And Gradio template.

**Comment by deep-diver** [comment]
Let's first merge this PR. I will open separate issues for each one of them

**Comment by sayakpaul** [comment]
Sounds good.

**Comment by deep-diver** [comment]
@sayakpaul 

oh, I didn't know this PR is still open. I thought we were going to open separate PRs for the listed issues after merging this PR.

**Comment by sayakpaul** [comment]
> Let's

https://github.com/deep-diver/semantic-segmentation-ml-pipeline/pull/38#issuecomment-1296301152

**Comment by deep-diver** [comment]
then, I will create a branch on top of the branch of this PR for each issues :)

**Comment by sayakpaul** [comment]
Do we want to merge it first?

**Comment by deep-diver** [comment]
that was my initial intention, wdyt?

**Comment by sayakpaul** [comment]
Sounds good.

**Pull request merged**